


// let promise = new Promise((resolve,reject)=>{

//     // resolve("resolved promise")
//     reject("rejected promise ")



// })

// promise.then((data)=>{
//     console.log(data);
//     return data;
// }).then((data)=>{
//     console.log(data);
//     return data;
// }).then((data)=>{
//     console.log(data);
//     return data;
// }).then((data)=>{
//     console.log(data);
//     return data;
// }).then((data)=>{
//     console.log(data);
//     return data;
// }).then((data)=>{
//     console.log(data);
//     return data;
// }).catch((error)=>{
//     console.log(error);
//     throw new Error("my custom error")
// }).catch((error)=>{
//     console.log(error);
//     throw new Error("my custom error")
   
// }).catch((error)=>{
//     console.log(error);
  
// })

// let promise = new Promise((resolve,reject)=>{
//     // resolve("resolved promise ")
//     // reject("rejected promise ")
//     // resolve("resolved pro0msie ")
//     reject("rejected pr0omsie ")

// })

// // promise.catch(error=>console.log(error))
// // .then(data=>console.log(data))
// // .then(data=>console.log(data))


// promise.then(data=>console.log(data))
// .catch(error=>console.log(error))


// let promise = new Promise((resolve , reject)=>{
//     // resolve("resolved data ")
//     // reject("rejected promise ")
//     // resolve("resolved promise")
//     // reject("rejected promise ")

// })

// // promise.then(data=>{
// //     console.log(data)
// // return data})
// // .then(data=>{
// //     console.log(data)
// // return data}
// // )
// // .then(data=>{
// //     console.log(data)
// //     return data
// // })
// // .catch(error=>console.log(error))

// promise.catch(error=>{
//     console.log(error);
//     throw new Error("custom error")
// }).catch(error=>{
//     console.log(error);
//     throw new Error("custom error")
// }).catch(error=>{
//     console.log(error);
//     throw new Error("custom error")
// })